tests/test_wps_split_merged_climos.py

```diff
@@ -6,16 +6,31 @@
 from .common import client_for, TESTDATA
 from thunderbird.processes.wps_split_merged_climos import SplitMergedClimos
 
-gcm_360_climos_local = TESTDATA["test_local_gcm_360_climos_nc"]
-gcm_climos_local = TESTDATA["test_local_gcm_climos_nc"]
-tasmax_climos_local = TESTDATA["test_local_tasmax_climos_nc"]
-gcm_360_climos_opendap = TESTDATA["test_opendap_gcm_360_climos_nc"]
-gcm_climos_opendap = TESTDATA["test_opendap_gcm_climos_nc"]
-tasmax_climos_opendap = TESTDATA["test_opendap_tasmax_climos_nc"]
+# limiting test_data to climo files
+split_merged_climos_local = [
+    nc for nc in TESTDATA["test_local_nc"] if nc.endswith("_climos.nc")
+]
+split_merged_climos_opendap = [
+    od for od in TESTDATA["test_opendaps"] if od.endswith("_climos.nc")
+]
 
 client = client_for(Service(processes=[SplitMergedClimos()]))
 
 
+def build_netcdf_sets(climo_files):
+    netcdf_sets = []
+
+    i = 0
+    while i < len(climo_files):
+        j = i + 1
+        while j < len(climo_files):
+            netcdf_sets.append((climo_files[i], climo_files[j]))
+            j += 1
+        i += 1
+
+    return netcdf_sets
```

Comment:
The intention of this function is to build a list of tuples of 2 climo files. Each tuple represents a combination that can be made from a list. For example, assume an argument is [file1, file2, file3]. We want a result to be something like this: [(file1, file2),(file1, file3),(file2, file3)].
> Looking at run_multiple_files it seems it would be happy with a list as input.

Yes, I believe it is true. You can put the whole list of bunch of climo files, and I believe that has to work(if not I think there is something wrong with the program dealing with multiple inputs). However, I just wanted to have many test cases. For the worst case testing, I think it would be nice to add a tuple with all of the climo files on top of the built list. What do you think @nikola-rados ?

Reply:
> We want a result to be something like this: [(file1, file2),(file1, file3),(file2, file3)].

Ok, I see where you were going with this.  You wrote a function that would create a bunch of cases that could be mapped in `pytest.mark.parametrize`.  It is clever, but I think it may add just a bit too much complexity. Would you be able to try building the tuple list from scratch instead of from an existing list? It may make it a bit clearer what is happening. Although I think we may want to just manually create a handful of cases to avoid confusion.

> However, I just wanted to have many test cases.

Many test cases is _fine_, but if they are not actually different cases they are not super useful. Imagine you have a method that does this:
```python3
def foo(bar):
    return bar + bar
```
Anything we throw into this method, the output will be the same. No literally, but there are no "cases" to test other than the base case. So passing in a bunch of cases where `bar` is parametrized as `1`, `2`, `3` and so on, wouldn't really target anything different in the method.  

Whereas this method:
```python3
def foo(bar):
    if bar > 0:
        return bar
    else:
        return bar + bar
```
Has a few more cases to consider.  We may want a case where `bar` is positive, one where it is negative and maybe one where it is `0`.  All these cases seek to target different parts of the method. So we get value out of our cases.  Does that make sense?